### PR TITLE
mds: add dentry lease metric and some improvement fixes

### PR DESCRIPTION
**Pull request by lxbsz**
This bases on @vshankar's previous metric PRs [1] and [2]. Tested with kclient patch series [3]'s V2.

This should rebase to the upstream after [1] and [2], the first 15 commits, get merged.

[1] https://github.com/ceph/ceph/pull/26004
[2] https://github.com/ceph/ceph/pull/32201
[3] https://patchwork.kernel.org/project/ceph-devel/list/?series=221131

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: debian/ceph-mgr.install [hunk 1]
```diff
@@ -21,6 +21,7 @@ usr/share/ceph/mgr/prometheus
 usr/share/ceph/mgr/rbd_support
 usr/share/ceph/mgr/restful
 usr/share/ceph/mgr/selftest
+tusr/share/ceph/mgr/stats
```
Comment (gmayyyha) [inline debian/ceph-mgr.install]: tusr?

Comment (lxbsz) [reply debian/ceph-mgr.install]: @gmayyyha 
Maybe the first vshankar's 12 commits are old one, I didn't sync them yet.
This PR just cares about the last 4 ones. It should be rebased later.

Thanks.

**Comment by gregsfortytwo** [comment]
@lxbsz do you want to update this?

**Comment by lxbsz** [comment]
> @lxbsz do you want to update this?

It needs to wait for the depending PRs to get merged. I will update it then and also together with the relating metric patch series in kclient.

Thanks

**Comment by lxbsz** [comment]
still waiting....

**Comment by lxbsz** [comment]
jenkins test make check

**Comment by vshankar** [comment]
depends on https://github.com/ceph/ceph/pull/29951

**Comment by lxbsz** [comment]
Will wait https://github.com/ceph/ceph/pull/29951

**Comment by batrick** [comment]
@vshankar please have another look.

File: src/mds/MDSPerfMetricTypes.h [hunk 2]
```diff
@@ -106,17 +131,20 @@ struct Metrics {
...
 
   DENC(Metrics, v, p) {
-    DENC_START(1, 1, p);
+    DENC_START(2, 1, p);
     denc(v.update_type, p);
     denc(v.cap_hit_metric, p);
     denc(v.read_latency_metric, p);
     denc(v.write_latency_metric, p);
     denc(v.metadata_latency_metric, p);
+    if (struct_v >= 2)
```
Comment (vshankar) [inline src/mds/MDSPerfMetricTypes.h:146]: nit: enclose withith braces

**Review by vshankar — CHANGES_REQUESTED** [review]
One minor nit -- looks good. thx @lxbsz

File: src/mds/MDSPerfMetricTypes.h (hunk 2 above)
Comment (lxbsz) [reply src/mds/MDSPerfMetricTypes.h]: Done, thanks @vshankar

**Review by vshankar — APPROVED** [review]

**Comment by lxbsz** [comment]
@vshankar A small fixing about the `CEPHFS_METRIC_FEATURES_ALL ` to support the `CLIENT_METRIC_TYPE_DENTRY_LEASE`.

**Comment by batrick** [comment]
https://pulpito.ceph.com/pdonnell-2020-11-05_00:20:13-fs-wip-pdonnell-testing-20201104.220526-distro-basic-smithi/

failures known

**Pull request merged**